torchgeo/trainers/segmentation.py

```diff
@@ -250,9 +349,13 @@ def validation_step(
         batch_size = x.shape[0]
         y_hat = self(x)
         loss = self.criterion(y_hat, y)
-        self.log('val_loss', loss, batch_size=batch_size)
+        self.log('val_loss', loss, batch_size=batch_size, on_epoch=True)
         self.val_metrics(y_hat, y)
-        self.log_dict(self.val_metrics, batch_size=batch_size)
+        self.log_dict(
+            {f'{k}': v for k, v in self.val_metrics.compute().items()},
```

Comment:
I believe this is for the class-wise metrics but is not needed in later versions of TorchMetrics as `ClasswiseWrapper` handles the reporting correctly.

We should definitely get rid of this as `compute()` reduces the metric, but this should happen at the end of each *epoch* not *step*.

Reply:
Dropping on_epoch, val defaults to on_step, as does train.

Re compute, reverted to
```
        self.val_metrics(y_hat, y)
        self.log_dict(
            self.val_metrics,
            batch_size=batch_size,
        )
```
and get 
```
ValueError: The `.compute()` return of the metric logged as 'val_Accuracy' must be a tensor. Found {'multiclassaccuracy_background': tensor(0.0009, device='cuda:0'), 'multiclassaccuracy_burned_area': tensor(0.9992, device='cuda:0')}
```

Reply:
It works as expected on v1.4.0. I remember I had this issue when before I upgraded versions on my project and got around it by basically doing `metrics(input, target)` and then looping over the `MetricCollection` object and unpacking any iterables I found. I think this allowed me to get rid of the `compute` call.

Reply:
I would be all for `v1.4.0` personally :-)

Reply:
Ok looks like we support 1.4.1?? https://github.com/microsoft/torchgeo/blob/8c02cec3801672947455db92f76537072b15dd4a/requirements/required.txt#L21

Reply:
https://github.com/microsoft/torchgeo/blob/main/requirements/README.md

Reply:
Does it work in 1.3 or 1.2? 1.4 is pretty new. We do require the latest and greatest kornia, so it's not impossible to require such a new version, just don't want to make torchgeo difficult to install for people stuck on older versions.

Reply:
@DimitrisMantas can you share an example implementation re https://github.com/microsoft/torchgeo/pull/2130#discussion_r1707124290